roles/terraform/tasks/storage/disk_Debian_setup.yml

```diff
@@ -19,6 +36,75 @@
   with_items:
     - "{{ ansible_devices.keys() | select('match','vd[b-z]') | list }}"
 
+# Create directory under /root to store keyfiles that
+# are going to be generated.
+- name: Create directory to store keyfiles (remote)
+  file:
+    path: /root/keyfiles
+    state: directory
+    mode: '0600'
+  with_together:
+    - "{{ data_disks.values() | list }}"
+    - "{{ ansible_devices.keys() | select('match','vd[b-z]') | sort | list }}"
+  when:
+    - "{{ item.0['encryption'] }}"
+
+# Generate local directory to store keyfiles.
+- name: Create directory to store keyfiles (local)
+  file:
+    path: "{{ hcl_deploy_path | expanduser }}/keyfiles"
+    state: directory
+    mode: '0755'
+  delegate_to: localhost
+  with_together:
+    - "{{ data_disks.values() | list }}"
+    - "{{ ansible_devices.keys() | select('match','vd[b-z]') | sort | list }}"
+    - "{{ data_disks.keys() | list }}"
+  when:
+    - "{{ item.0['encryption'] }}"
+
+# Generate keyfiles using /dev/urandom and store them 
+# into a local directory
+- name: Generate keyfiles
+  shell: |
```

Comment:
shell must be idempotent, use exit codes and `changed_when` to ensure an `ok` when nothing is changed

Reply:
solved